transcom/mymove
# B-21458 Main

## [B-21458](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1015567)

## Summary
Agility: [B-21458](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1015567)
Update emails for Move Submitted, Move Counseled, and the PPM Payment Reminder.
When the orders type is retirement OR separation, the emails should state the destination address of their first shipment instead of Duty Station.
For PPMs, we need to use the Address linked to the PPM.
I also moved the formatting into Address.go to keep it consistent.

Int PR 1: https://github.com/transcom/mymove/pull/14099
Int PR 2: https://github.com/transcom/mymove/pull/14272

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Do the code changes match the INT pull requests?

### How to test

1. To enable email, run the server with: EMAIL_BACKEND=ses aws-vault exec transcom-gov-dev -- make server_run
2. Log in as a Customer with a valid email address so you can check emails.
3. Create a Retirement or Separation PPM, with a destination address different from the HOS/HOR address, with an expected departure date at least 14 days in the past.
4. You will receive a "Thank you for submitting your move details" email showing the actual destination address (instead of HOS/ HOR/PLEAD)
5. Log in as a Counselor
6. Approve the PPM
7. You will receive a "Your counselor has approved your move details" email showing the actual destination address (instead of HOS/ HOR/PLEAD)

Part 2 (I've only tested this in LoadTest, but it may work locally if you leave your server running over night):
8. The next morning you will receive a "Complete your Personally Procured Move (PPM)" email showing the actual destination address (instead of HOS/ HOR/PLEAD)

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
#### Submitted
![image](https://github.com/user-attachments/assets/e05bcee8-91ce-4d04-9a07-e249018b1a9a)
#### Approved
![image](https://github.com/user-attachments/assets/a48010c3-b54f-4c63-9857-c6c3e6ec4ad5)
##### Payment Reminder
(hasn't sent yet)
